File: chdsr/chdsr-simulator-web/src/main/java/io/iochord/dev/chdsr/simulator/web/v1/models/WebServiceResponse.java
package io.iochord.dev.chdsr.simulator.web.v1.models;

import lombok.Getter;
import lombok.Setter;

/**
 *
 * @package chdsr-simulator-web
 * @author  Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since   2019
 *
 *
 */
public class WebServiceResponse<T> {

	@Getter
	@Setter
	private T data;

	@Getter
	@Setter
	private WebServiceStatus status = WebServiceStatus.completed();
	
	public WebServiceResponse() {
		
	}
	
	public WebServiceResponse(T data) {
		setData(data);
	}
	
}

File: ips/ips-simulator-web/src/main/ts/src/iochord/ips/common/service/WebServiceResponse.ts
import { WebServiceStatus } from './WebServiceStatus';

/**
 * The web socket response class
 *
 * @export
 * @class WebServiceResponse
 *
 * @package ts
 * @author  Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since   2019
 */
export class WebServiceResponse {
  /**
   * Data response from the web socket.
   *
   * @type {*}
   * @memberof WebServiceResponse
   */
  public data: any = null;

  /**
   * The status of web socket response.
   *
   * @type {WebServiceStatus}
   * @memberof WebServiceResponse
   */
  public status: WebServiceStatus = new WebServiceStatus();
}
